test/test_torch.py

```diff
@@ -7986,10 +7986,12 @@ def _test_flip(self, use_cuda=False):
         self.assertRaises(RuntimeError, lambda: data.flip(0, 1, 1))
         # not allow empty list as input
         self.assertRaises(TypeError, lambda: data.flip())
+
         # not allow size of flip dim > total dims
-        self.assertRaises(RuntimeError, lambda: data.flip(0, 1, 2, 3))
+        exc = RuntimeError if use_cuda else IndexError
```

Comment:
why does this differ on cpu vs cuda?  Can you fix that?

Reply:
Yes, this one is in `aten/src/ATen/native/TensorTransformations.h`.  There are other candidate locations for conversion to `IndexError` like for example `aten/src/ATen/TensorOperators.h`:

```
  AT_CHECK(
    index.isIntegral(),
      "Can only index tensors with integral scalars");
```

Would you prefer all of them in one PR?

Sometimes the choice of exception is not obvious, for example in the above `data.flip(0, 1, 2, 3)` case `ValueError` or `TypeError` might be appropriate as well.

On the other hand it's probably easier if users just have to catch a single exception for all cases where "something indexing related" goes wrong.

Reply:
I don't think all of them need to be in the same PR, but it's strange if the behavior isn't the same on CPU and CUDA.